### ISSUE TITLE
Angular-Fullstack Update

### ISSUE THREAD

**Issue by drpalaric** [stormpath/stormpath-sdk-angularjs#100]
I think that the tutorial needs to be updated to the newest version of Angular-Fullstack. There are options in version 3 that are not in version 2. For example, you now have to choose between Mongo or SQL for a database, when before you could just not select anything. Also, now you have to select the testing suite that you will use, which wasn't there before. So, I think this probably should be updated. Thoughts?

**Comment by surfjedi** [comment]
+1

**Comment by robertjd** [comment]
Thanks everyone, is this a simple dependency update, or do we have to do more work?

**Comment by drpalaric** [comment]
Looks like dependency updates. I believe the Angular tutorial probably needs to be updated to reflect what the new Yeoman options are regarding the database and testing suite.

**Comment by surfjedi** [comment]
If any one is interested I have done and port to webpack build for this example:
https://github.com/berzerk-interactive/express-stormpath-angular-sample-project/tree/feature/webpack

**Comment by surfjedi** [comment]
And it has a lot of dependency updates, including node and npm.  FYI I have been using new node 5.9 successfully too

**Comment by lcollet** [comment]
What about Stormpath SDK with another backend than ExpressJS?
I got some weird behaviours using it with a rest api based on SpringBoot

**Comment by drpalaric** [comment]
@lcollet Here is Stormpath's Spring Boot example https://github.com/stormpath/stormpath-spring-boot-tutorial

**Comment by lcollet** [comment]
@AJAlger yeah, i know, my backend is a Java one (using SpringBoot for local dev, and Wildfly for production).
But i got some issues using it with the Stormpath Angular SDK.
The Stormpath js lib tells me that it cannot get a "view model" from backend (for login and register pages).
By default, SpringBoot backend with Stormpath comes with default mappings ('/login', 'register', ...).
As i can see a POST request for /login exists, but no a GET one (i got a 404)?
According to the spec, it should be : https://github.com/stormpath/stormpath-framework-spec/blob/master/login.md
Do you understand what i mean?

**Comment by chanoch** [comment]
@lcollet You need to provide your own login page if you want this to work. The framework simply provides the end point for authentication which is the POST to /login

Can you be more specific about the weird behaviours if this doesn't cover your issue?

**Comment by lcollet** [comment]
@chanoch 
So, the specs is only fully implemented for a JS backend (Node, Express)?
So, i cannot use the "dynamic login JSON view model"? It's not explicit when you read the docs (https://docs.stormpath.com/angularjs/sdk/#/api).
I have to write the login JSON view model myself to be able to use the login page, alright?
Maybe i should take a look to this repo : https://github.com/stormpath/stormpath-jersey-sample.

Another question: should i omit the use of the AngularJS SDK then?
I'll upload a picture of the behaviour of the login page later.

Thank u guys!

**Comment by robertjd** [comment]
Hello @lcollet !  As you've found in our framework spec, the /login route should return a JSON response on GET requests, this response is the [login view model](https://github.com/stormpath/stormpath-framework-spec/blob/master/login.md#login-view-model), which tells you what social providers are configured.  The registration endpoint has a similar behavior, the [registration view model](https://github.com/stormpath/stormpath-framework-spec/blob/master/registration.md#registration-view-model) gives information about the fields that you have configured for the registration form.

Our Java team is currently working on adding these features to the Java SDK, you can track the progress with this issue:

https://github.com/stormpath/stormpath-sdk-java/issues/364

If you need more information about timeline I would try them on that issue. If you need an urgent solution, you can return a JSON object for the JSON GET requests on these routes.  You can simply copy from the examples in the framework spec and they should work.

Please let me know if I can offer any assistance, you can also contact us through support@stormpath.com.  Thank you for the questions!

**Comment by lcollet** [comment]
@robertjd Thank you for your help !
I already opened a support ticket! All the answers i got should allow me to implement my own solution as you suggest it.
Guys, i really appreciate your support.

**Comment by robertjd** [comment]
Oh awesome, always happy to help!
